app/models/miq_server/worker_management/monitor/validation.rb

```diff
@@ -4,22 +4,50 @@ module MiqServer::WorkerManagement::Monitor::Validation
   def validate_worker(w)
     return true if MiqEnvironment::Command.is_podified?
 
-    time_threshold   = get_time_threshold(w)
-    memory_threshold = get_memory_threshold(w)
+    if !worker_heartbeat_valid?(w)
+      stop_worker(w, MiqServer::NOT_RESPONDING)
+      return false
+    end
 
-    w.validate_active_messages
+    if !worker_memory_valid?(w)
+      stop_worker(w)
```

Comment:
No reason?

Reply:
This is how it was previously [ref](https://github.com/ManageIQ/manageiq/pull/19609/files#diff-57a2568ae37208b39c60baa82b4c3a9cL38)

Reply:
weird...probably should be a TODO for later

Reply:
So interestingly it looks like only 2 of the 5 callers provide a reason:
```
app/models/miq_server/worker_management/monitor.rb:127:      stop_worker(w, MiqServer::MEMORY_EXCEEDED)
app/models/miq_server/worker_management/monitor/quiesce.rb:42:        stop_worker(w)
app/models/miq_server/worker_management/monitor/validation.rb:8:      stop_worker(w, MiqServer::NOT_RESPONDING)
app/models/miq_server/worker_management/monitor/validation.rb:15:      stop_worker(w)
app/models/miq_worker.rb:405:    miq_server.stop_worker_queue(self)
```

Would have to look more into what we do with the reasons, but that might be too shiny for right now.